java/src/jmri/jmrix/ieee802154/xbee/XBeeIOStream.java

```diff
@@ -153,7 +153,7 @@ public void run() { // start a new thread
                    dataArray[i++] = n;
                 }
                 if (log.isDebugEnabled()) {
-                    log.debug("XBee Thread received message " + dataArray);
+                    log.debug("XBee Thread received message " + jmri.util.StringUtil.hexStringFromBytes(dataArray));
```

Comment:
While in this case, the expensive part of logging is the ```hexStringFromBytes``` call and this call is protected by the isDebugEnabled() test, changing
```
log.debug("XBee Thread received message " + jmri.util.StringUtil.hexStringFromBytes(dataArray));
```
to
```
log.debug("XBee thread received message {}" , jmri.util.StringUtil.hexStringFromBytes(dataArray));
```
is generally a faster way of not logging (if, for example, the lcf disables logging at all) than building the logged String as a parameter passed to ```.debug(...)```